CHANGES.md

```diff
@@ -1,6 +1,433 @@
+  (see https://chapel-lang.org/docs/master/modules/standard/Random.html#Random.makeRandomStream)
+* added type query routines for 'owned', 'shared', 'unmanaged', and 'borrowed'
+  (see https://chapel-lang.org/docs/master/modules/standard/Types.html#Types.isOwnedClass)
+* made the Sync `Barrier` reusable
+* increased the use of error-handling in standard modules
+* converted some non-fatal checks from `halt()` to `warning()`
+* extended string methods to support UTF-8 strings
+* added array.front() and array.back() to get the first/last element of an array
+* improved array.pop_front() and array.pop_back() to return the removed element
+
+Package Modules
+---------------
+* added an HDF5 module for reading/writing files using the HDF5 interface
+  (see TODO)
+* added a NetCDF module for reading/writing files using the NetCDF interface
+* converted `TOML` module to use error-handling
+  (see https://chapel-lang.org/docs/master/modules/packages/TOML.html#TOML.TomlError)
+* added buffered versions of non-fetching atomic operations
+  (see https://chapel-lang.org/docs/master/modules/packages/BufferedAtomics.html)
+* replaced strings with enumerations for cipher variations in the Crypto module
+
+Standard Domain Maps (Layouts and Distributions)
+------------------------------------------------
+
+New Tools / Tool Changes
+------------------------
+* added support for non-Chapel system packages in mason
+  (see https://chapel-lang.org/docs/master/tools/mason/mason.html#using-system-dependencies)
+* added support for non-Chapel Spack packages in mason
+  (see https://chapel-lang.org/docs/master/tools/mason/mason.html#using-spack-dependencies)
+* added `mason add` and `mason rm` for managing dependencies
+  (see https://chapel-lang.org/docs/master/tools/mason/mason.html#using-chapel-dependencies)
+* added support for tests within mason
+  (see https://chapel-lang.org/docs/master/tools/mason/mason.html#testing-your-package)
+* added support for examples within mason
+  (see https://chapel-lang.org/docs/master/tools/mason/mason.html#creating-and-running-examples)
+* added `mason run --release` flag to specify release vs. debug
+* added `--no-update` flag for better offline support in mason
+* added support for calling mason commands from any directory within project
+* added support for checking last modified timestamp before building in mason
+* added 'chpldoc' support for new keywords: 'overrides', 'owned, 'borrowed', ...
+* squashed 'chpldoc' documentation of compiler-generated symbols
+* added Python 3 support for chpldoc
+* improved bash tab completion to handle developer flags in developer mode only
+* fixed a bash tab completion bug for paths involving '~'
+
+Interoperability Improvements
+-----------------------------
+* added initial support for exporting Chapel code for use as a Python library
+  (see https://chapel-lang.org/docs/latest/technotes/libraries.html)
+  (TODO: I may have a more specific link once I've updated the document)
```

Comment:
https://chapel-lang.org/docs/master/technotes/libraries.html#using-your-library-in-python - is the better Python link I have created